libsolidity/formal/CHC.cpp

```diff
@@ -923,8 +918,10 @@ void CHC::defineInterfacesAndSummaries(SourceUnit const& _source)
 						auto state1 = stateVariablesAtIndex(1, *base);
 						auto state2 = stateVariablesAtIndex(2, *base);
 
-						auto nondetPre = iface(state0 + state1);
-						auto nondetPost = iface(state0 + state2);
+						auto const* iface = m_nondetInterfaces.at(base);
+						auto state0 = stateVariablesAtIndex(0, *base);
+						auto nondetPre = (*iface)(state0 + state1);
+						auto nondetPost = (*iface)(state0 + state2);
```

Comment:
Personally, for me (probably anyone else looking this for the first time) it is too hard to follow all the types here when over-using auto. Anyway, I am fine with it.

Reply:
I agree with you that at a first glance that might be the case, but when you're used to this file you know that these are Predicates applications (smtutil::Expression). In any case, I'm also fine with either and will change if you think it makes sense.

Reply:
I am fine with leaving auto